integration-tests/src/test/java/com/datastax/oss/driver/core/SerializationIT.java

```diff
@@ -0,0 +1,142 @@
+import com.datastax.oss.driver.api.testinfra.session.SessionRule;
+import com.datastax.oss.driver.api.testinfra.simulacron.SimulacronRule;
+import com.datastax.oss.simulacron.common.cluster.ClusterSpec;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.RuleChain;
+import org.junit.rules.TestRule;
+
+public class SerializationIT {
+  private static final SimulacronRule SIMULACRON_RULE =
+      new SimulacronRule(ClusterSpec.builder().withNodes(1));
+
+  private static final SessionRule<CqlSession> SESSION_RULE =
+      SessionRule.builder(SIMULACRON_RULE).build();
+
+  @ClassRule
+  public static final TestRule CHAIN = RuleChain.outerRule(SIMULACRON_RULE).around(SESSION_RULE);
+
+  @Before
+  public void clear() {
+    SIMULACRON_RULE.cluster().clearPrimes(true);
+  }
+
+  @Test
+  public void should_serialize_node() {
+    // Given
+    Node node = SESSION_RULE.session().getMetadata().getNodes().values().iterator().next();
+
+    // When
+    Node deserializedNode = serializeAndDeserialize(node);
+
+    // Then
+    // verify a few fields, no need to be exhaustive
+    assertThat(deserializedNode.getHostId()).isEqualTo(node.getHostId());
+    assertThat(deserializedNode.getEndPoint()).isEqualTo(node.getEndPoint());
+    assertThat(deserializedNode.getCassandraVersion()).isEqualTo(node.getCassandraVersion());
+  }
+
+  @Test
+  public void should_serialize_driver_exception() {
+    // Given
+    SIMULACRON_RULE.cluster().prime(when("mock query").then(serverError("mock server error")));
+    try {
+      SESSION_RULE.session().execute("mock query");
+      fail("Expected a ServerError");
+    } catch (ServerError error) {
+      assertThat(error.getExecutionInfo()).isNotNull();
+
+      // When
+      ServerError deserializedError = serializeAndDeserialize(error);
+
+      // Then
+      assertThat(deserializedError.getMessage()).isEqualTo("mock server error");
+      assertThat(deserializedError.getCoordinator().getEndPoint())
+          .isEqualTo(error.getCoordinator().getEndPoint());
+      assertThat(deserializedError.getExecutionInfo()).isNull(); // transient
+    }
+  }
+
+  @Test
+  public void should_serialize_row() {
+    // Given
+    SIMULACRON_RULE
+        .cluster()
+        .prime(when("mock query").then(rows().row("t", "mock data").columnTypes("t", "varchar")));
+    Row row = SESSION_RULE.session().execute("mock query").one();
+
+    // When
+    row = serializeAndDeserialize(row);
+
+    // Then
+    ColumnDefinition columnDefinition = row.getColumnDefinitions().get("t");
+    assertThat(columnDefinition.getType()).isEqualTo(DataTypes.TEXT);
+    assertThat(row.getString("t")).isEqualTo("mock data");
+  }
+
+  // Serialization methods copied from SerializationHelper in core tests.
+
+  private static <T> byte[] serialize(T t) {
```

Comment:
It's the 3rd time already that we are copying-pasting this code: I see one `SerializationHelper` in native-protocol, another one in OSS core. Why don't you re-use the one from OSS core? The test-jar is already being produced, it's just a matter of adding a dependency.

Reply:
I got ahead of myself a little bit on this one: I think when the two drivers get merged, we should stop deploying a `core` test JAR. Maven central is a community resource and we should use as little of it as possible. Publishing a test JAR for every release just to factor a few lines of trivial code is not a good reason IMHO.

But right now we have to publish it anyway because it's used in the DSE driver, which reuses more than a few trivial things I think. I'll add the dependency as you suggest, and we can discuss this again in the driver merge PR.